assets/js/googlesitekit/datastore/user/surveys.test.js

```diff
@@ -0,0 +1,125 @@
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	createTestRegistry,
+} from '../../../../../tests/js/utils';
+import {
+	act,
+} from '../../../../../tests/js/test-utils';
+import { STORE_NAME } from './constants';
+
+describe( 'core/user surveys', () => {
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	const survey = {
+		survey_payload: 'foo',
+		session: 'bar',
+	};
+
+	describe( 'actions', () => {
+		describe( 'triggerSurvey', () => {
+			it( 'throws an error when parameters are missing or incorrect', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey();
+				} ).toThrow( 'triggerID is mandatory and must be a string' );
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey( 'ABCD', false );
+				} ).toThrow( 'options must be an object' );
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey( 'ABCD', { ttl: 'a' } );
+				} ).toThrow( 'options.ttl must be a number' );
+			} );
+
+			it( 'does not throw an error when parameters are correct', () => {
+				fetchMock.post(
+					/^\/google-site-kit\/v1\/core\/user\/data\/survey-trigger/,
+					{ body: survey, status: 200 }
+				);
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey( 'a' );
+				} ).not.toThrow();
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey( 'b', { ttl: 1 } );
+				} ).not.toThrow();
+			} );
+		} );
+
+		describe( 'sendSurveyEvent', () => {
+			it( 'throws an error when parameters are missing or incorrect', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).sendSurveyEvent();
+				} ).toThrow( 'eventID is mandatory and must be a string' );
+				expect( () => {
+					registry.dispatch( STORE_NAME ).sendSurveyEvent( 'a', 'b' );
+				} ).toThrow( 'eventData must be an object' );
+			} );
+
+			it( 'does not throw an error when parameters are correct', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).sendSurveyEvent( 'a' );
+				} ).not.toThrow();
+				expect( () => {
+					registry.dispatch( STORE_NAME ).sendSurveyEvent( 'b', { foo: 'bar' } );
+				} ).not.toThrow();
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey( 'c' );
+					registry.dispatch( STORE_NAME ).sendSurveyEvent( 'd', { foo: 'bar' } );
+				} ).not.toThrow();
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getCurrentSurvey', () => {
+			it( 'returns null when no current survey is set', async () => {
+				expect( registry.select( STORE_NAME ).getCurrentSurvey() ).toEqual( null );
```

Comment:
So you could do this with `expect( foo ).toBeNull();` because
* IMHO it's more readable
* `toEqual` actually does a deep comparison